### PR TITLE
Lb2 docs

### PR DESCRIPTION
**Pull request by wprzytula**
This are the changes & amendments in small commits for easier review. I'll rebase them after approval.

### CODE REVIEW

File: scylla/src/transport/load_balancing/plan.rs [hunk 1]
```diff
@@ -13,6 +13,12 @@ enum PlanState<'a> {
     },
 }
 
+/// The list of nodes constituting the query plan.
+///
+/// The plan is partly lazily computed, with the first node computed
+/// instantly and the remaining nodes computed on-demand (all at once).
```
Comment (havaker) [inline scylla/src/transport/load_balancing/plan.rs]: Using a word "instantly" is a bit missleading here? It sounds like if that calculation would occur during construction - on `new()`, but tje first node is computed on the first `.next()` invocation.

File: scylla/src/transport/load_balancing/mod.rs [hunk 2]
```diff
@@ -35,6 +35,10 @@ pub struct RoutingInfo<'a> {
     pub is_confirmed_lwt: bool,
 }
 
+/// The list of nodes in the query plan not including the most preferred node.
```
Comment (havaker) [inline scylla/src/transport/load_balancing/mod.rs]: Fallback plan may include the most preferred node, it is filtered later by `Plan`.

File: docs/source/load-balancing/default-policy.md [hunk 3]
```diff
@@ -94,6 +151,8 @@ The DefaultPolicy prefers to return nodes in the following order:
 3. Alive local nodes
 4. Alive remote nodes (if datacenter failover is permitted & possible due to consistency constraints)
 5. Enabled down nodes
+And only if latency awareness is enabled:
+6. Penalised: alive local replicas, alive remote replicas, ... (in order as above).
```
Comment (havaker) [inline docs/source/load-balancing/default-policy.md]: I don't fully understand this.

Our load balancing plan is a composition of the `pick` result with the `fallback` result.
In the `pick` method we are using the predicate generated by `LatencyAwareness`, so we won't return penalized nodes (except some edge cases). In the `fallback` method, we are using `Self::is_alive` predicate instead, so we are free to return penalized nodes. Therefore the latency-awareness affects only the first element of the load balancing plan.

Reading this documentation fragment, I had the impression that:
- penalized nodes are added to the very end of the plan
- there is more than one penalized node in plan. 

However, as explained above, it is not the case with this policy.

**Review by havaker — CHANGES_REQUESTED** [review]

File: scylla/src/transport/load_balancing/mod.rs (hunk 2 above)
Comment (wprzytula) [reply scylla/src/transport/load_balancing/mod.rs]: Good catch.

File: docs/source/load-balancing/default-policy.md (hunk 3 above)
Comment (wprzytula) [reply docs/source/load-balancing/default-policy.md]: Please take a closer look at `default.rs:173`:
```
if let Some(latency_awareness) = self.latency_awareness.as_ref()
    Box::new(latency_awareness.wrap(plan))
} else {
    Box::new(plan)
}
```
...and at `wrap()` definition:
```
pub(super) fn wrap<'a>(
    &self,
    fallback: impl Iterator<Item = NodeRef<'a>>,
) -> impl Iterator<Item = NodeRef<'a>> {
    let min_avg_latency = match self.last_min_latency.load() {
        Some(min_avg) => min_avg,
        None => return Either::Left(fallback), // noop, as no latency data has been collected yet
    };

    Either::Right(IteratorWithSkippedNodes::new(
        self.node_avgs.read().unwrap().deref(),
        fallback,
        self.exclusion_threshold,
        self.retry_period,
        self.minimum_measurements,
        min_avg_latency,
    ))
}
```
Moreover, the tests I've written confirm that the policy behaves according to my assumptions.

Do you agree now that every node in the plan is subject to penalisation?

Comment (havaker) [reply docs/source/load-balancing/default-policy.md]: Thanks for the explanation! As you noticed I missed wrapping at the end of `fallback`.

> Do you agree now that every node in the plan is subject to penalisation?

I do agree with that, I also withdraw my remarks from the previous comment;)

**Review by havaker — APPROVED** [review]

**Pull request merged**